sbncode/SBNEventWeight/App/SystToolsEventWeight_module.cc

```diff
@@ -0,0 +1,238 @@
+
+        systtools::event_unit_response_t resp = *itResp;
+        //==== resp.size() corresponds to number of knobs we altered;
+        //==== e.g., MaCCQE, MaCCRES, MvCCRE -> resp.size() = 3
+        if(fDebugMode){
+          std::cout << "[SystToolsEventWeight::produce]     sp->GetSystMetaData().size() (expected) = " << sp->GetSystMetaData().size() << "\n"
+                    << "[SystToolsEventWeight::produce]     resp.size() of this syst_resp (produced) = " << resp.size() << std::endl;
+        }
+        if(sp->GetSystMetaData().size()!=resp.size()){
+          std::cerr << "[SystToolsEventWeight::produce]     sp->GetFullyQualifiedName() = " << sp->GetFullyQualifiedName() << std::endl;
+          std::cerr << "[SystToolsEventWeight::produce]     We expect to have " << sp->GetSystMetaData().size() << " knobs for this SystProvider, but "
+                    << resp.size() << " are produced. "
+                    << "Probably this particular event is not relevant to this systematic variation." << std::endl;
+          throw std::exception();
+        }
+
+        sbn::evwgh::EventWeightMap mcwgh;
+
+        for( systtools::event_unit_response_t::iterator it = resp.begin(); it != resp.end(); ++it ){
+
+          //==== responses.size() is the number of universes
+          systtools::SystParamHeader const &sph = fParamHeaderHelper.GetHeader( (*it).pid );
+          std::string prettyName = sph.prettyName;
+
+          if(fDebugMode){
+            std::cout << "[SystToolsEventWeight::produce]       pid of this resp = " << (*it).pid << "\n"
+                      << "[SystToolsEventWeight::produce]       prettyName of this resp = " << prettyName << "\n"
+                      << "[SystToolsEventWeight::produce]       paramVariations.size() of this resp (expected) = " << sph.paramVariations.size() << "\n"
+                      << "[SystToolsEventWeight::produce]       responses.size() of this resp (produced) = " << (*it).responses.size() << std::endl;
+          }
+          if(sph.paramVariations.size()!=(*it).responses.size()){
+            std::cerr << "[SystToolsEventWeight::produce]       prettyName of this resp = " << prettyName << std::endl;
+            std::cerr << "[SystToolsEventWeight::produce]       We expect to have " << sph.paramVariations.size() << " universes, but "
+                      << (*it).responses.size() << " are produced. "
+                      << "Probably this particular event is not relevant to this systematic variation." << std::endl;
+            throw std::exception();
+          }
+
+          std::vector<float> wgts;
+          for( unsigned int i = 0; i < (*it).responses.size(); ++i ){
+
+            if(fDebugMode) std::cout << "[SystToolsEventWeight::produce]         (*it).responses[i] = " << (*it).responses[i] << std::endl;
+            wgts.push_back( (*it).responses[i] );
+
+          } // END loop over universes of this knob
+
+          mcwgh[sp->GetFullyQualifiedName()+"_"+prettyName] = wgts;
```

Comment:
Is this just a copy, like:
```suggestion
          mcwgh[sp->GetFullyQualifiedName()+"_"+prettyName] = r.responses;
```
or is there any implicit conversion? in that case:
```suggestion
          mcwgh[sp->GetFullyQualifiedName()+"_"+prettyName].assign(r.responses.cbegin(), r.responses.cend());
```
or am I missing anything?

Reply:
`r.responses` is `std::vector<double>` (https://github.com/LArSoft/systematicstools/blob/d6d620acfc2b3e6eae14489ac1454594cb3a97f6/systematicstools/interface/EventResponse_product.hh#L39) while the values of `mcwgh` are `std::vector<float>` (https://github.com/SBNSoftware/sbnobj/blob/de8ec694d7b1c824518e4232a603bf260f234f45/sbnobj/Common/SBNEventWeight/EventWeightMap.h#L18). Better use the later one?

Reply:
Updated as suggested; followed the later case.